kafka-plugins-0.10/src/main/java/io/cdap/plugin/batch/source/KafkaBatchSource.java

```diff
@@ -134,16 +135,16 @@ public KafkaBatchSource(Kafka10BatchConfig config) {
 
   @Override
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
-    config.validate();
-    pipelineConfigurer.getStageConfigurer().setOutputSchema(config.getSchema());
+    config.validate(pipelineConfigurer.getStageConfigurer().getFailureCollector());
+    pipelineConfigurer.getStageConfigurer().setOutputSchema(config.getSchema(pipelineConfigurer.getStageConfigurer().getFailureCollector()));
   }
 
   @Override
   public void prepareRun(BatchSourceContext context) throws Exception {
     Job job = JobUtils.createInstance();
     Configuration conf = job.getConfiguration();
 
-    KafkaPartitionOffsets partitionOffsets = config.getInitialPartitionOffsets();
+    KafkaPartitionOffsets partitionOffsets = config.getInitialPartitionOffsets(context.getFailureCollector());
```

Comment:
if there was an error, need to throw an exception before continuing.

```
context.getFailureCollector().getOrThrowException();
```

This is different than what happens at configurePipeline() time, because here it's doing actual work that requires valid values. At configure time, all that is happening is other validation logic.

These things are validated again at prepareRun() even though they have already been validated at configure() time because any property annotated with `@Macro` can be set to a placeholder that has no value at configure() time, but has a value here at runtime.